ql/src/java/org/apache/hadoop/hive/ql/exec/vector/mapjoin/fast/VectorMapJoinFastBytesHashKeyRef.java

```diff
@@ -69,12 +69,14 @@ public static int calculateHashCode(long refWord, WriteBuffers writeBuffers,
 
       // And, if current value is big we must read it.
       actualKeyLength = writeBuffers.readVInt(readPos);
-      keyAbsoluteOffset = absoluteOffset + actualKeyLength;
+
+      // Now the read position is set to start of the key as readVInt moved the
+      // position by size of key length.
+      return writeBuffers.hashCode(actualKeyLength, readPos);
```

Comment:
I guess the fact that we have perform a read to get the actual KeyLen is making things more complex here. 
Shall we add some comments on them method level for future ref?